src/format/locales.rs

```diff
@@ -31,3 +31,115 @@ pub(crate) fn d_t_fmt(locale: Locale) -> &'static str {
 pub(crate) fn t_fmt(locale: Locale) -> &'static str {
     locale_match!(locale => LC_TIME::T_FMT)
 }
+
+macro_rules! expand_ampm {
+    ($item:ident) => {
+        pure_rust_locales::$item::LC_TIME::T_FMT_AMPM
+    };
+}
+
+/// hardcoded `locale_match!` fallback for `T_FMT_AMPM`.
+///
+/// it takes variable number of possible locales that have `LC_TIME::T_FMT_AMPM` field.
+/// it will then match the locale and return its `LC_TIME::T_FMT_AMPM` field.
+/// fallbacks to en_US if the field is an empty string.
+///
+/// ```rs
+/// // usage:
+/// locale => ( Locale::POSIX, Locale::aa_DJ, ...)
+/// // expands into:
+/// match locale {
+///     Locale::POSIX => {
+///         let ampm = pure_rust_locales::POSIX::LC_TIME::T_FMT_AMPM;
+///
+///         if ampm.is_empty() { pure_rust_locales::en_US::LC_TIME::T_FMT_AMPM } else { ampm }
+///     },
+///     Locale::aa_DJ => // ...
+///     // ...
+///     _ => pure_rust_locales::en_US::LC_TIME::T_FMT_AMPM,
+/// }
+/// ```
+macro_rules! locale_match_ampm {
+    ($locale:expr => ( $($item:ident),* $(,)? )) => {
+        match $locale {
+            $(Locale::$item => {
+                let ampm = expand_ampm!($item);
+
+                if ampm.is_empty() { expand_ampm!(en_US) } else { ampm }
+            },)*
+
+            // default fallback is en_US
+            _ => expand_ampm!(en_US),
+        }
+    }
+}
+
+/// finds locale's time format in 12 hour AM/PM.
+///
+/// uses hardcoded [locale_match_ampm] macro to find `LC_TIME::T_FMT_AMPM` field.
+///
+/// ## Why locale_match won't work
+///
+/// [locale_match] assumes every single locale module to have same field.
+/// however, due to issues in [pure-rust-locales][issue],
+/// three locales (ff_SN, km_KH, ug_CN) are missing `LC_TIME::T_FMT_AMPM` field,
+/// causing compile error.
+///
+/// example:
+/// ```rs
```

Comment:
This docstring code should be ignored. Also, I don't think `rs` is valid

    /// ```rust,ignore